src/main/java/Main.java

```diff
@@ -0,0 +1,33 @@
+import dao.CartDao;
+import dao.ProductDao;
+import entity.Cart;
+import entity.Product;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+
+public class Main {
+    public static void main(String[] args) {
+        AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
+        context.register(DatabaseConfig.class);
+        context.scan("dao");
+        context.refresh();
```

Comment:
Instead of manually scanning specific packages, let's enable automatic component scanning provided by Spring out of the box. To do so, you should specify `@ComponentScan` annotation in `DatabaseConfig` class and pass the packages you need to scan.